mono/metadata/threads.c

```diff
@@ -568,33 +568,37 @@ set_current_thread_for_domain (MonoDomain *domain, MonoInternalThread *thread, M
 }
 
 static MonoThread*
-create_thread_object (MonoDomain *domain)
+create_thread_object (MonoDomain *domain, MonoError *error)
 {
-	MonoError error;
 	MonoVTable *vt = mono_class_vtable (domain, mono_defaults.thread_class);
-	MonoThread *t = (MonoThread*)mono_object_new_mature (vt, &error);
-	mono_error_raise_exception (&error); /* FIXME don't raise here */
+	MonoThread *t = (MonoThread*)mono_object_new_mature (vt, error);
 	return t;
 }
 
 static MonoThread*
 new_thread_with_internal (MonoDomain *domain, MonoInternalThread *internal)
 {
-	MonoThread *thread = create_thread_object (domain);
+	MonoError error;
+	MonoThread *thread;
+
+	thread = create_thread_object (domain, &error);
+	mono_error_raise_exception (&error); /* FIXME don't raise here */
+
 	MONO_OBJECT_SETREF (thread, internal_thread, internal);
+
 	return thread;
 }
 
 static MonoInternalThread*
-create_internal_thread (void)
+create_internal_thread (MonoError *error)
```

Comment:
It is already split in 2 commits, `create_internal_thread` is modified in https://github.com/ludovic-henry/mono/commit/71c1a8ad9befbc7391018c74e4e771fb3dfb1ee9